cmd/convert.go

```diff
@@ -0,0 +1,176 @@
+		ConvertFromFile(filePath, graphFormat, imageFormat)
+	}
+}
+
+func ConvertFromStdin(diagramType string, imageFormat string) {
+	if diagramType == "" {
+		exit("diagram type must be specify using --type flag")
+	}
+	text, err := GetTextFromStdin()
+	if err != nil {
+		exit(err)
+	}
+	result, err := client.FromString(text, kroki.Graphviz, kroki.ImageFormat(imageFormat))
+	if err != nil {
+		exit(err)
+	}
+	fmt.Println(result)
+}
+
+func GetTextFromStdin() (result string, err error) {
+	reader := bufio.NewReader(os.Stdin)
+	input, err := ioutil.ReadAll(reader)
+	return string(input), err
+}
+
+func ConvertFromFile(filePath string, graphFormatRaw string, imageFormatRaw string) {
+	graphFormat, err := ResolveGraphFormat(graphFormatRaw, filePath)
+	if err != nil {
+		exit(err)
+	}
+	imageFormat, err := ResolveImageFormat(imageFormatRaw)
+	if err != nil {
+		exit(err)
+	}
+	result, err := client.FromFile(filePath, graphFormat, imageFormat)
+	if err != nil {
+		exit(err)
+	}
+	err = client.WriteToFile(OutputFilePath(filePath, imageFormat), result)
+	if err != nil {
+		exit(err)
+	}
+}
+
+func OutputFilePath(filePath string, imageFormat kroki.ImageFormat) string {
+	fileExtension := path.Ext(filePath)
+	return filePath[0:len(filePath)-len(fileExtension)] + "." + string(imageFormat)
+}
+
+func ResolveImageFormat(imageFormatRaw string) (result kroki.ImageFormat, err error) {
+	value := strings.ToLower(imageFormatRaw)
+	switch value {
+	case "svg":
+		return kroki.Svg, nil
+	case "png":
+		return kroki.ImageFormat("png"), nil
+	case "jpeg":
+		return kroki.ImageFormat("jpeg"), nil
+	case "pdf":
+		return kroki.ImageFormat("pdf"), nil
+	default:
+		return kroki.ImageFormat(""), errors.Errorf(
+			"invalid image format %s.",
+			value)
+	}
+}
+
+func ResolveGraphFormat(graphFormatRaw string, filePath string) (result kroki.GraphFormat, err error) {
+	if graphFormatRaw == "" {
+		return GraphFormatFromFile(filePath)
+	} else {
+		return GraphFormatFromValue(graphFormatRaw)
+	}
+}
+
+func GraphFormatFromValue(value string) (result kroki.GraphFormat, err error) {
+	value = strings.ToLower(value)
+	switch value {
+	case "dot", "graphviz":
+		return kroki.Graphviz, nil
+	case "plantuml":
+		return kroki.Plantuml, nil
+	case "nomnoml":
+		return kroki.Nomnoml, nil
+	case "blockdiag":
+		return kroki.BlockDiag, nil
+	case "mermaid":
+		return kroki.Mermaid, nil
+	case "svgbob":
+		return kroki.Svgbob, nil
+	case "umlet":
+		return kroki.Umlet, nil
+	case "c4plantuml":
+		return kroki.C4plantuml, nil
+	case "seqdiag":
+		return kroki.SeqDiag, nil
+	case "erd":
+		return kroki.GraphFormat("erd"), nil
+	case "nwdiag":
+		return kroki.GraphFormat("nwdiag"), nil
+	case "actdiag":
+		return kroki.GraphFormat("actdiag"), nil
+	case "ditaa":
+		return kroki.GraphFormat("ditaa"), nil
+	default:
+		return kroki.GraphFormat(""), errors.Errorf(
+			"invalid graph format %s.",
+			value)
+	}
+}
+
+func GraphFormatFromFile(filePath string) (result kroki.GraphFormat, err error) {
```

Comment:
You don't need to name the return values, `(kroki.GraphFormat, error)` is enough.

Reply:
I thought it was mandatory, removed!
Named return values is "just" for documentation purpose ?

Reply:
Yes, it's for documentation: https://tour.golang.org/basics/7